stream_alert/athena_partition_refresh/main.py

```diff
@@ -271,11 +292,69 @@ def repair_hive_table(self, unique_buckets):
 
         return True
 
-    @staticmethod
-    def add_hive_partition(_):
-        """Execute a Hive Add Partition command on a given Athena table"""
-        LOGGER.error('Add Hive Parition is not yet supported, exiting!')
-        raise NotImplementedError
+    def add_hive_partition(self, s3_buckets_and_keys):
+        """Execute a Hive Add Partition command on a given Athena table
+
+        Args:
+            s3_buckets_and_keys (dict): Buckets and unique keys to add partitions
+
+        Returns:
+            (bool): If the repair was successful for not
+        """
+        athena_config = self.config['lambda']['athena_partition_refresh_config']
+        add_hive_partition_config = athena_config['refresh_type']['add_hive_partition']
+        partitions = set()
+
+        for bucket, keys in s3_buckets_and_keys.iteritems():
+            athena_table = add_hive_partition_config.get(bucket)
+            if not athena_table:
+                LOGGER.error('%s not found in add_hive_partition config. '
+                             'Please add this bucket to enable additions '
+                             'of Hive partitions.',
+                             athena_table)
+                continue
+
+            # Gather all of the partitions to add per bucket
+            s3_key_regex = self.STREAMALERTS_REGEX if athena_table == 'alerts' else self.FIREHOSE_REGEX
+            for key in keys:
+                match = s3_key_regex.search(key)
+                if len(match.groups()) != 4:
+                    LOGGER.warning('%s does not match the regex: %s',
+                                   key, s3_key_regex.pattern)
+                    continue
+
+                # Convert the match groups to a dict for easy access
+                match_dict = match.groupdict()
+                # Get the path to the objects in S3
+                path = '/'.join(key.split('/')[:-1])
```

Comment:
Is the goal here to just strip everything after and including the last '/'? If so, and if I'm understanding this correctly, maybe consider `os.path.dirname(key)` or `os.path.split(key)[0]`